lib/command.c

```diff
@@ -1322,8 +1323,16 @@ int command_config_read_one_line(struct vty *vty,
 		}
 	}
 
-	if (ret != CMD_SUCCESS && ret != CMD_WARNING)
-		memcpy(vty->error_buf, vty->buf, VTY_BUFSIZ);
+	if (ret != CMD_SUCCESS && ret != CMD_WARNING) {
+		struct vty_error *ve = XCALLOC(MTYPE_TMP, sizeof(*ve));
+
+		memcpy(ve->error_buf, vty->buf, VTY_BUFSIZ);
```

Comment:
I know this was here before but it's kinda janky, this is a string buffer and we're doing a memcpy for the full size of the buffer without a null terminator? and then in the printout later it looks like we replace newlines with `\0` and i guess that makes it work? this really should be
```snprintf(vy->error_buf, VTY_BUFSIZ, "%s", vty->buf)```